client/components/wcc-settings-form/index.js

```diff
@@ -15,64 +15,84 @@ import ObjectPath from 'objectpath';
 import * as SettingsActions from 'state/settings/actions';
 import * as PackagesActions from 'state/form/packages/actions';
 
-const WCCSettingsForm = ( props ) => {
-	const {
-		layout,
-		saveFormData,
-		formActions,
-		noticeActions,
-	} = props;
-
-	const setIsSaving = ( value ) => formActions.setField( 'isSaving', value );
-	const setSuccess = ( value ) => {
-		formActions.setField( 'success', value );
+const WCCSettingsForm = React.createClass( {
```

Comment:
Since now the component needs lifecycle methods (`componentDidMount`, `componentWillUnmount`), it can't be a functional component anymore. So it's either this, or ES6 classes (which I've read somewhere that are fundamentally broken).